lib/synapse/config_generator/haproxy.rb

```diff
@@ -792,8 +795,6 @@ class Haproxy
     STATE_FILE_UPDATE_INTERVAL = 60.freeze # iterations; not a unit of time
 
     def initialize(opts)
-      super()
-
       %w{global defaults reload_command}.each do |req|
         raise ArgumentError, "haproxy requires a #{req} section" if !opts.has_key?(req)
       end
```

Comment:
Not related to this PR, but we should make sure we fix the ordering of req_pairs and setting default values for do_writes/do_socket/do_reloads. I'll fix that in the next iteration.